homeassistant/components/advantage_air/number.py

```diff
@@ -0,0 +1,63 @@
+"""Number platform for Advantage Air integration."""
+from typing import Any
+
+from homeassistant.components.number import NumberEntity
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import TIME_MINUTES
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN as ADVANTAGE_AIR_DOMAIN
+from .entity import AdvantageAirAcEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up AdvantageAir number platform."""
+
+    instance = hass.data[ADVANTAGE_AIR_DOMAIN][config_entry.entry_id]
+
+    entities = []
+    for ac_key in instance["coordinator"].data["aircons"]:
+        entities.append(AdvantageAirTimeTo(instance, ac_key, "On"))
+        entities.append(AdvantageAirTimeTo(instance, ac_key, "Off"))
+    async_add_entities(entities)
+
+
+class AdvantageAirTimeTo(AdvantageAirAcEntity, NumberEntity):
+    """Representation of Advantage Air TimeTo number."""
+
+    _attr_native_unit_of_measurement = TIME_MINUTES
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_native_step = 1
+    _attr_native_min_value = 0
+    _attr_native_max_value = 720
+
+    def __init__(self, instance: dict[str, Any], ac_key: str, action: str) -> None:
+        """Initialize the Advantage Air timer number."""
+        super().__init__(instance, ac_key)
+        self._time_key = f"countDownTo{action}"
+        self._attr_name = f"Time to {action}"
+        self._attr_unique_id = (
+            f'{self.coordinator.data["system"]["rid"]}-{self.ac_key}-timeto{action}'
+        )
+
+    @property
+    def native_value(self):
+        """Return the current value."""
+        return self._ac.get(self._time_key)
```

Comment:
It seems to me, that this value will actually be counting down / decreasing.
That is something that is not allowed.

Reply:
It replaces a sensor entity which is counting down. So while that specific issue/rule isnt being addressed by this PR, it does improve the user experience.

It seems HA doesnt have as solution for handling relative timers then? Should this PR instead remove this user functionality rather than improve it?

Reply:
> It replaces a sensor entity which is counting down.

That is allowed.

>  So while that specific issue/rule isnt being addressed by this PR, it does improve the user experience.

That might be in this case, hard to judge for me, however, that doesn't change the fact that we don't allow for this construct.

> In the past someone said to use an absolute datetime, but thats going to have twice as many entity updates as it will flip between two absolute datetimes every minute.

That is correct and the best solution. If it flips, it means you have a bug or are using an incorrect/non-matching resolution/precision.

../Frenck